packages/rum-core/src/domain/assembly.ts

```diff
@@ -79,15 +79,16 @@ export function startRumAssembly(
     ({ startTime, rawRumEvent, domainContext, savedCommonContext, customerContext }) => {
       const viewContext = parentContexts.findView(startTime)
       const urlContext = urlContexts.findUrl(startTime)
-      if (session.isTracked(startTime) && viewContext && urlContext) {
+      const session = sessionManager.findSession(startTime)
+      if (session && viewContext && urlContext) {
```

Comment:
Is there any cases where we can have a session and no viewContext or no urlContext?

Reply:
I'd say no

Reply:
To me this check is misleading. I would be happy if we could improve it. 
For intance:
```js
if (session) {
  const viewContext = parentContexts.findView(startTime)
  const urlContext = urlContexts.findUrl(startTime)
  ...
}
```
Wdyt?

Reply:
It should not be the case but it is not a guarantee that we have by design. 
Your proposal can't be added as is, we should either:

1.  still have to check if viewContext and urlContext are defined
2. add `!` to inform ts about that it should be defined at this point

For 1, I'd say it does not look that great and we would have to do it at several places, I'd prefer to stay with the current implem
For 2, in case I'm wrong about the initial question, I'd prefer to not introduce potential unrelated behavior changes with all pending changes on session. We could mitigate that but I don't think it worth the effort.

Another approach could be to have a design that is better expressing this state, I think we could revisit that when it will be a bit more bothering.

Reply:
Indeed, it does not have to be mixed with the current PR. 
I'm a bit concerned though. Aren't we trading the meaning of the code to please the linter?

To me, the code should reflect the behaviour we intend from the SDK. 
If we fear an unwanted behaviour, we could add an internal monitoring like for any other investigation. 

Of course, another design could better reflect the state, but if we are always exprecting  `findView` and `findUrl` to return a value, why not doing:

findUrl: (startTime?: RelativeTime) => UrlContext **~~| undefined~~**
findView: (startTime?: RelativeTime) => ViewContext **~~| undefined~~**

We can talk about it outside the PR if you want.